ocs_ci/ocs/ui/provider_client_ui.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import time
+
+from ocs_ci.ocs.ui.page_objects.page_navigator import PageNavigator
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs.resources.storageconsumer import (
+    get_all_storageconsumer_names,
+    StorageConsumer,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class StorageClientUI(PageNavigator):
+    """
+    User Interface Selenium for Storage Clients page of Provider cluster UI
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def generate_new_token(self):
+        """
+        Create a new client onboarding token in the UI
+
+        Return:
+            string: onboarding token
+
+        """
+        self.navigate_client_page()
+        logger.info("Click on 'Generate client onboarding token'")
+        self.do_click(self.validation["generate_token"])
+        token = self.get_element_text(self.validation["token"])
+        return token
+
+    def verify_client_data_in_ui(self):
+        """
+        Verify client details on Storage Clients page
+
+        """
+        self.navigate_client_page()
+        consumer_names = get_all_storageconsumer_names()
+        for consumer_name in consumer_names:
+            consumers_obj = ocp.OCP(
+                kind=constants.STORAGECONSUMER,
+                namespace=config.cluster_ctx.ENV_DATA["cluster_namespace"],
+                resource_name=consumer_name,
+            )
+            client_name = consumers_obj.get("status").get("client").get("name")
+            logger.info(f"Search for {client_name} client")
+            self.do_send_keys(self.validation["search_client"], text=client_name)
```

Comment:
can't find here that we clean-up input value.
If no clean-up we will get next client_name merged with previous